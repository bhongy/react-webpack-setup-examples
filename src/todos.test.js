// @flow strict
import React from 'react';
import { shallow, type ShallowWrapper } from 'enzyme';
import Todos from './todos';

// TODO: extract to a test helper module
type MockEventTarget = {
  name?: string,
  value: string,
};
const mockEvent = (target?: MockEventTarget = { value: '' }) => ({
  preventDefault() {},
  currentTarget: target,
});

describe('Todos', () => {
  const enterInput = (input: ShallowWrapper, value: string): void => {
    const event = mockEvent({ value });
    input.simulate('change', event);
  };

  it('does not render empty list', () => {
    const todos = shallow(<Todos />);
    expect(todos).toMatchSnapshot();
  });

  it('disables submit button when input is empty', () => {
    const todos = shallow(<Todos />);
    expect(todos.find('button').prop('disabled')).toEqual(true);
  });

  it('enables submit button when input has value', () => {
    const todos = shallow(<Todos />);
    const input = todos.find('input');
    enterInput(input, 'anything');
    expect(todos.find('button').prop('disabled')).toEqual(false);
  });

  it('renders the list of values', () => {
    const value = 'Hasenschwanzbruch';

    const todos = shallow(<Todos />);
    const input = todos.find('input');
    enterInput(input, value);
    expect(todos).toMatchSnapshot();
    expect(todos.state('inputValue')).toEqual(value);

    const form = todos.find('form');
    form.simulate('submit', mockEvent());
    expect(todos).toMatchSnapshot();
    expect(todos.state('inputValue')).toEqual('');
    expect(todos.state('values')).toEqual([value]);
  });
});
