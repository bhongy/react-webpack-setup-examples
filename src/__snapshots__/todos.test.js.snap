// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todos does not render empty list 1`] = `
<React.Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="value to add"
      type="text"
      value=""
    />
    <button
      disabled={false}
    >
      Add
    </button>
  </form>
</React.Fragment>
`;

exports[`Todos renders the list of values 1`] = `
<React.Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="value to add"
      type="text"
      value="Hasenschwanzbruch"
    />
    <button
      disabled={true}
    >
      Add
    </button>
  </form>
</React.Fragment>
`;

exports[`Todos renders the list of values 2`] = `
<React.Fragment>
  <ul>
    <li
      key="0"
    >
      Hasenschwanzbruch
    </li>
  </ul>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="value to add"
      type="text"
      value=""
    />
    <button
      disabled={false}
    >
      Add
    </button>
  </form>
</React.Fragment>
`;
